tests/test_leaks.py

```diff
@@ -23,3 +23,20 @@ def test_client_response_does_not_leak_on_server_disconnected_error() -> None:
         stdout=subprocess.PIPE,
     ) as proc:
         assert proc.wait() == 0, "ClientResponse leaked"
+
+
+@pytest.mark.skipif(IS_PYPY, reason="gc.DEBUG_LEAK not available on PyPy")
+def test_request_does_not_leak_when_request_handler_raises() -> None:
+    """Test that the Request object is collected when the handler raises.
+
+    https://github.com/aio-libs/aiohttp/issues/10548
+    """
+    leak_test_script = pathlib.Path(__file__).parent.joinpath(
+        "isolated", "check_for_request_leak.py"
```

Comment:
The only change from the above test is the filename. Should we just make a parametrized test (maybe even getting the filenames automatically from the directory)?

Reply:
Good idea

Reply:
https://github.com/aio-libs/aiohttp/pull/10577